modules/ml/src/svmsgd.cpp

```diff
@@ -0,0 +1,202 @@
+//     and/or other materials provided with the distribution.
+//
+//   * The name of the copyright holders may not be used to endorse or promote products
+//     derived from this software without specific prior written permission.
+//
+// This software is provided by the copyright holders and contributors "as is" and
+// any express or implied warranties, including, but not limited to, the implied
+// warranties of merchantability and fitness for a particular purpose are disclaimed.
+// In no event shall the Intel Corporation or contributors be liable for any direct,
+// indirect, incidental, special, exemplary, or consequential damages
+// (including, but not limited to, procurement of substitute goods or services;
+// loss of use, data, or profits; or business interruption) however caused
+// and on any theory of liability, whether in contract, strict liability,
+// or tort (including negligence or otherwise) arising in any way out of
+// the use of this software, even if advised of the possibility of such damage.
+//
+//M*/
+
+#include "precomp.hpp"
+#include "svmsgd.hpp"
+
+/****************************************************************************************\
+*                        Stochastic Gradient Descent SVM Classifier                      *
+\****************************************************************************************/
+
+namespace cv {
+namespace ml {
+
+SVMSGD::SVMSGD(float lambda, float learnRate, uint nIterations){
+
+    // Initialize with random seed
+    _randomNumber = 1;
+
+    // Initialize constants
+    _slidingWindowSize = 0;
+    _nFeatures = 0;
+    _predictSlidingWindowSize = 1;
+
+    // Initialize sliderCounter at index 0
+    _sliderCounter = 0;
+
+    // Parameters for learning
+    _lambda = lambda;  // regularization
+    _learnRate = learnRate;  // learning rate (ideally should be large at beginning and decay each iteration)
+    _nIterations = nIterations;  // number of training iterations
+
+    // True only in the first predict iteration
+    _initPredict = true;
+
+    // Online update flag
+    _onlineUpdate = false;
+}
+
+SVMSGD::SVMSGD(uint updateFrequency, float lambda, float learnRate, uint nIterations){
+
+    // Initialize with random seed
+    _randomNumber = 1;
+
+    // Initialize constants
+    _slidingWindowSize = 0;
+    _nFeatures = 0;
+    _predictSlidingWindowSize = updateFrequency;
+
+    // Initialize sliderCounter at index 0
+    _sliderCounter = 0;
+
+    // Parameters for learning
+    _lambda = lambda;  // regularization
+    _learnRate = learnRate;  // learning rate (ideally should be large at beginning and decay each iteration)
+    _nIterations = nIterations;  // number of training iterations
+
+    // True only in the first predict iteration
+    _initPredict = true;
+
+    // Online update flag
+    _onlineUpdate = true;
+
+    // Learn rate decay: _learnRate = _learnRate * _learnDecay
+    _learnRateDecay = 0.1;
```

Comment:
_learnRateDecay will be defined by an input variable, like _predictSlidingWindowSize